services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py

```diff
@@ -310,30 +310,39 @@ async def task_containers_restart(
     settings: ApplicationSettings,
     shared_store: SharedStore,
 ) -> None:
-    progress.update(message="starting containers restart", percent=0.0)
-    if shared_store.compose_spec is None:
-        raise RuntimeError("No spec for docker-compose command was found")
+    assert app.state.container_inspect_lock  # nosec
 
-    for container_name in shared_store.container_names:
-        await stop_log_fetching(app, container_name)
+    # NOTE: if containers inspect reports that the containers are restarting
+    # or some other state, the service will get shutdown, to prevent this
+    # blocking status while containers are being restarted.
+    async with app.state.container_inspect_lock:
```

Comment:
Ok, I see now the reason. Then I have a suggestion which is the following:
- what about renaming that lock to something like ```container_restart_lock``` ?
- the action that happens when you want to prevent the inspection is that you restart, not that you inspect right?

Also side question, is your lock not going to block a burst of calls to the inspect endpoint?

Reply:
Will rename, makes it clearer.

So if I get the question right: the lock will prevent a burst of calls to the inspect lock from going though, it will only handle one call at a time. Which I guess is OK.

Reply:
just that if the director-v2 is calling this every 5 seconds, and with retries, it will block until it restarted... is this ok?

Reply:
The director-v2 will schedule a call to the onbservation cycle every 5 seconds. While the observation cycle is running the container inspect will be called and retried. 
After the inspect is over (with or without retries) a new observation cycle can be scheduled.

So you only have at most 1 other call being blocked by the lock.